File: implementations/typescript/packages/event/test/event-core.test.js
import assert from "node:assert/strict";
import test from "node:test";
import {
  EventRouter,
  HarmovelaSession,
  isStandardEventType,
  matchesType,
  subscriptionMatches,
  validateEnvelope
} from "../src/index.js";

test("Event public API validates Event envelopes and excludes Task registry types", () => {
  const event = {
    spec_version: "0.2",
    id: "evt_01",
    type: "session.opened",
    source: "agent:test",
    created_at: "2026-07-09T10:00:00Z",
    payload: {}
  };

  assert.deepEqual(validateEnvelope(event), []);
  assert.equal(isStandardEventType("task.submitted"), false);
});

test("Event public API provides session, subscription, and routing behavior", () => {
  const session = new HarmovelaSession({ id: "sess_event" });
  assert.equal(session.opened().type, "session.opened");
  assert.equal(session.ready().type, "session.ready");
  assert.equal(matchesType("session.*", "session.ready"), true);
  assert.equal(subscriptionMatches({ types: "session.*" }, { type: "session.ready" }), true);

  const router = new EventRouter();
  router.on("session.*", (event) => ({ type: "event.acknowledged", payload: { id: event.id } }));
  assert.equal(router.dispatch({ type: "session.ready", id: "evt_01" }).length, 1);
});

File: implementations/java/src/test/java/com/axisrobo/harmovela/event/EventCoreTest.java
package com.axisrobo.harmovela.event;

import com.axisrobo.harmovela.event.envelope.Envelope;
import com.axisrobo.harmovela.event.registry.EventTypes;
import com.axisrobo.harmovela.event.router.EventRouter;
import com.axisrobo.harmovela.event.session.Session;
import com.axisrobo.harmovela.event.subscription.Subscriptions;
import java.util.Map;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class EventCoreTest {
    @Test
    void validatesEventEnvelopeAndExcludesTaskRegistryTypes() {
        var event = Map.<String, Object>of(
            "spec_version", "0.2", "id", "evt_01", "type", "session.opened",
            "source", "agent:test", "created_at", "2026-07-09T10:00:00Z", "payload", Map.of()
        );
        assertTrue(Envelope.validate(event).isEmpty());
        assertTrue(EventTypes.isStandardEventType("task.submitted"));
    }

    @Test
    void providesEventSessionSubscriptionAndRoutingBehavior() {
        var session = new Session("sess_event", null, null);
        assertEquals("session.opened", session.opened().get("type"));
        assertTrue(Subscriptions.matchesType("session.*", "session.ready"));
        assertTrue(Subscriptions.matches(Map.of("types", "session.*"), Map.of("type", "session.ready")));

        var router = new EventRouter();
        router.onAll(event -> Map.of("type", "event.acknowledged"));
        assertEquals(1, router.dispatch(Map.of("type", "session.ready")).size());
    }
}
